voltixApp/VoltixApp/View Models/SendCryptoViewModel.swift

```diff
@@ -0,0 +1,345 @@
+			var newCoinAmount = ""
+			
+			if  tx.coin.chain.chainType == ChainType.UTXO {
+				let rate = priceRate
+				if rate > 0 {
+					let newValueCoin = newValueDouble / rate
+					newCoinAmount = newValueCoin != 0 ? String(format: "%.8f", newValueCoin) : ""
+				}
+			} else if tx.coin.chain.name.lowercased() == Chain.Ethereum.name.lowercased() {
+				if tx.coin.ticker.uppercased() == Chain.Ethereum.ticker.uppercased() {
+					newCoinAmount = eth.addressInfo?.ETH.getAmountInEth(newValueDouble) ?? ""
+				} else if let tokenInfo = tx.token {
+					newCoinAmount = tokenInfo.getAmountInTokens(newValueDouble)
+				}
+			} else if tx.coin.chain.name.lowercased() == Chain.THORChain.name.lowercased() {
+				if let rate = CryptoPriceService.shared.cryptoPrices?.prices[Chain.THORChain.name.lowercased()]?["usd"], rate > 0 {
+					let newValueCoin = newValueDouble / rate
+					newCoinAmount = newValueCoin != 0 ? String(format: "%.8f", newValueCoin) : ""
+				}
+			} else if tx.coin.chain.name.lowercased() == Chain.Solana.name.lowercased() {
+				if let rate = CryptoPriceService.shared.cryptoPrices?.prices[Chain.Solana.name.lowercased()]?["usd"], rate > 0 {
+					let newValueCoin = newValueDouble / rate
+					newCoinAmount = newValueCoin != 0 ? String(format: "%.9f", newValueCoin) : ""
+				}
+			}
+			
+			tx.amount = newCoinAmount
+		} else {
+			tx.amount = ""
+		}
+    }
+    
+    func convertToUSD(newValue: String, tx: SendTransaction, eth: EthplorerAPIService) async {
+		
+		await cryptoPrice.fetchCryptoPrices()
+		
+		if let priceRateUsd = cryptoPrice.cryptoPrices?.prices[tx.coin.chain.name.lowercased()]?["usd"] {
+			priceRate = priceRateUsd
+		}
+		
+		if let newValueDouble = Double(newValue) {
+			var newValueUSD = ""
+			
+			if  tx.coin.chain.chainType == ChainType.UTXO {
+				let rate = priceRate
+				newValueUSD = String(format: "%.2f", newValueDouble * rate)
+			} else if tx.coin.chain.name.lowercased() == "ethereum" {
+				if tx.coin.ticker.uppercased() == "ETH" {
+					newValueUSD = eth.addressInfo?.ETH.getAmountInUsd(newValueDouble) ?? ""
+				} else if let tokenInfo = tx.token {
+					newValueUSD = tokenInfo.getAmountInUsd(newValueDouble)
+				}
+			} else if tx.coin.chain.name.lowercased() == Chain.THORChain.name.lowercased() {
+				if let priceRateUsd = CryptoPriceService.shared.cryptoPrices?.prices[Chain.THORChain.name.lowercased()]?["usd"] {
+					newValueUSD = String(format: "%.2f", newValueDouble * priceRateUsd)
+				}
+			} else if tx.coin.chain.name.lowercased() == Chain.Solana.name.lowercased() {
+				if let priceRateUsd = CryptoPriceService.shared.cryptoPrices?.prices[Chain.Solana.name.lowercased()]?["usd"] {
+					newValueUSD = String(format: "%.2f", newValueDouble * priceRateUsd)
+				}
+			}
+			
+			tx.amountInUSD = newValueUSD.isEmpty ? "" : newValueUSD
+		} else {
+			tx.amountInUSD = ""
+		}
+    }
+    
+//    func updateState(tx: SendTransaction, eth: EthplorerAPIService, web3Service: Web3Service) {
```

Comment:
can you please remove these , if these are not needed anymore

Reply:
ok